tdrs-backend/tdpservice/reports/views.py

```diff
@@ -0,0 +1,52 @@
+"""Define API views for reports app."""
+
+from wsgiref.util import FileWrapper
+from django.http import FileResponse
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+
+from tdpservice.reports.models import ReportFile
+from tdpservice.reports.serializers import ReportFileSerializer, ReportIngestSerializer
+from tdpservice.users.permissions import IsApprovedPermission, ReportFilePermissions
+
+
+class ReportFileViewSet(ModelViewSet):
+    """Report file views."""
+
+    http_method_names = ["get", "post", "head"]
+    queryset = ReportFile.objects.all().order_by("-created_at")
+    serializer_class = ReportFileSerializer
+    permission_classes = [ReportFilePermissions, IsApprovedPermission]
+
+    def get_serializer_context(self):
+        """Retrieve additional context required by serializer."""
+        context = super().get_serializer_context()
+        context["user"] = self.request.user
+        return context
+
+    @action(methods=["post"], detail=False)
+    def master(self, request, pk=None):
```

Comment:
- When testing the /reports/master endpoint I always get a 400 regardless of the file I upload
<img width="1655" height="554" alt="Screenshot 2025-10-30 at 3 21 36 PM" src="https://github.com/user-attachments/assets/f61048d7-f381-4850-8c97-5749be784f8d" />

Reply:
This should be resolved.

Reply:
As a approved non-admin data analyst I am able to post a master.zip to the master endpoint. When I do, I get the following stack trace:

```
ERROR Internal Server Error: /v1/reports/master/
Traceback (most recent call last):
File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
response = get_response(request)
File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 204, in _get_response
response = response.render()
File "/usr/local/lib/python3.10/site-packages/sentry_sdk/integrations/django/views.py", line 41, in sentry_patched_render
return old_render(self)
File "/usr/local/lib/python3.10/site-packages/django/template/response.py", line 105, in render
self.content = self.rendered_content
File "/usr/local/lib/python3.10/site-packages/rest_framework/response.py", line 70, in rendered_content
ret = renderer.render(self.data, accepted_media_type, context)
File "/usr/local/lib/python3.10/site-packages/rest_framework/renderers.py", line 724, in render
context = self.get_context(data, accepted_media_type, renderer_context)
File "/usr/local/lib/python3.10/site-packages/rest_framework/renderers.py", line 655, in get_context
raw_data_post_form = self.get_raw_data_form(data, view, 'POST', request)
File "/usr/local/lib/python3.10/site-packages/rest_framework/renderers.py", line 541, in get_raw_data_form
if not self.show_form_for_method(view, method, request, instance):
File "/usr/local/lib/python3.10/site-packages/rest_framework/renderers.py", line 433, in show_form_for_method
view.check_object_permissions(request, obj)
File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 345, in check_object_permissions
if not permission.has_object_permission(request, self, obj):
File "/tdpapp/tdpservice/users/permissions.py", line 193, in has_object_permission
return user_stt == obj.stt_id
AttributeError: 'ReportIngest' object has no attribute 'stt_id
```

Reply:
The zip was also successfully processed despite the stack trace.

Reply:
```suggestion
    def ingest(self, request, pk=None):
```
Keeping with the removal of the 'master' paradigm

Reply:
> Keeping with the removal of the 'master' paradigm

Just pushed a full refactor on the ReportIngest model to name it ReportSource. So all `ingest` and `master` verbiage is now `report_source`.

Looking into the permission issue you noted above now

Reply:
This should be resolved now. Non-admin shouldn't have access to GET or POST on `v1/report/report-sources/`